extensions/websockets-next/kotlin/src/main/kotlin/io/quarkus/websockets/next/runtime/kotlin/CoroutineInvoker.kt

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.websockets.next.runtime.kotlin
+
+import io.quarkus.arc.Arc
+import io.smallrye.mutiny.Uni
+import io.smallrye.mutiny.coroutines.asUni
+import io.vertx.core.Vertx
+import jakarta.enterprise.invoke.Invoker
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+
+object CoroutineInvoker {
+    @JvmStatic
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun <T, U> inNewCoroutine(instance: T, arguments: Array<Any?>, invoker: Invoker<T, U>): Uni<U> {
+        val coroutineScope = Arc.container().instance(ApplicationCoroutineScope::class.java).get()
+        val dispatcher: CoroutineDispatcher =
```

Comment:
I'm sorry for my ignorance but what exactly is this `dispatcher` used for? I kind of miss the whole picture of how a kotlin continuation fits the current "Execution model" (as described in the javadoc and docs, e.g. [here](https://quarkus.io/version/main/guides/websockets-next-reference#invocation-rules)).

Reply:
This is how the `suspend` functions are executed on the Vert.x event loop.

Reply:
I actually think that the request context activation in `VertxDispatcher` is not necessary, but testing session context is a good idea. Will do.

Reply:
Done.